client/my-sites/term-tree-selector/add-term.jsx

```diff
@@ -193,6 +207,15 @@ class TermSelectorAddTerm extends Component {
 							onKeyUp={ this.boundValidateInput } />
 						{ isError && <FormInputValidation isError text={ this.state.error } /> }
 					</FormFieldset>
+					{ showDescriptionInput && <FormFieldset>
+							<FormLegend>
+								{ translate( 'Description', { context: 'Terms: Term description label' } ) }
+							</FormLegend>
+							<FormTextarea
```

Comment:
I updated the component to handle non-hierarchical taxonomies. I think we can consider merging for now. and we will handle editing once we have the listing.